Bio/SeqFeature.py

```diff
@@ -1188,6 +1198,24 @@ def __iter__(self):
             for pos in loc:
                 yield pos
 
+    def __eq__(self, other):
+        """Check if all parts of CompoundLocation are equal to all parts of other CompoundLocation"""
+        if not isinstance(other, CompoundLocation):
+            return False
+        if not len(self.parts) == len(other.parts):
+            return False
+        if not self.operator == other.operator:
+            return False
+        identical = True
+        for i, self_part in enumerate(self.parts):
```

Comment:
I'd prefer the zip option here. Note just use ``zip`` as in this context it will work fine under Python 2 (it not be an iterator but the list memory overhead is not important).